python/grass/temporal/space_time_datasets.py

```diff
@@ -662,11 +661,10 @@ def get_np_array(self):
         array back into grass.
         """
 
-        a = garray.array3d()
-
         if self.map_exists():
-            a.read(self.get_map_id())
-
+            garray.array3d(self.get_map_id())
```

Comment:
`a = ` is missing, but it might be better to just do direct return here anyway. Same for 667 and in RasterDataset on 240 and 242.